pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -681,7 +681,7 @@ func PublishAppInstMetaDataToZedCloud(ctx *zedagentContext, appInstID string, ap
 	}
 	statusURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "info")
 
-	deferKey := "appInstMetadataInfo:" + appInstID
+	deferKey := "appInstMetadataInfo:" + pubsubKeyStr
```

Comment:
You can concatenate the type string in here to make it be unique.

Reply:
It seems that would not work, as there is also [publishAppInstMetadata](https://github.com/lf-edge/eve/blob/410c6a65972b03f2e72443376aedf056b3ba8e90/pkg/pillar/cmd/zedrouter/zedrouter.go#L1044), where concatenation would also be needed. And 4 more places actually

Reply:
> It seems that would not work, as there is also [publishAppInstMetadata](https://github.com/lf-edge/eve/blob/410c6a65972b03f2e72443376aedf056b3ba8e90/pkg/pillar/cmd/zedrouter/zedrouter.go#L1044), where concatenation would also be needed. And 4 more places actually

Correct. I think Milan was suggesting to set the ReportAppInstMetaData.Uuid to the UUID in the AppInstMetaData and keep the pubsubkey as in the PR.

Reply:
done